src/go/transform-sdk/sr/client.go

```diff
@@ -14,5 +14,230 @@
 
 package sr
 
+import (
+	"encoding/binary"
+	"errors"
+	"strconv"
+	"unsafe"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk/internal/rwbuf"
+)
+
 // schemaId is an ID of a schema registered with schema registry
 type schemaId uint32
+
+// SchemaType is an enum for the different types of schemas that can be stored in schema registry.
+type SchemaType int
+
+const (
+	TypeAvro SchemaType = iota
+	TypeProtobuf
+	TypeJSON
+)
+
+// SchemaReference is a way for a one schema to reference another. The details
+// for how referencing is done are type specific; for example, JSON objects
+// that use the key "$ref" can refer to another schema via URL. For more details
+// on references, see the following link:
+//
+//	https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#schema-references
+//	https://docs.confluent.io/platform/current/schema-registry/develop/api.html
+type Reference struct {
+	Name    string
+	Subject string
+	Version int
+}
+
+// Schema is a schema that can be registered within schema registry
+type Schema struct {
+	Schema     string
+	Type       SchemaType
+	References []Reference
+}
+
+// SchemaSubject is a schema along with the subject, version and ID of the schema
+type SubjectSchema struct {
+	Schema
+
+	Subject string
+	Version int
+	ID      int
```

Comment:
I wonder if using fixed size types here would reduce the number of int/int32 casts in the code.

Reply:
Generally go only uses variable sized types, but the wasm ABI code requires fixed size types, so I think to give our users a nice API we use the variable sized ones, and do the casts ourself.

FWIW we use `int32` in many places here but golang's `int` type is almost always 8 bytes.